scilpy/denoise/tests/test_bilateral_filtering.py

```diff
@@ -0,0 +1,57 @@
+import numpy as np
+
+from scilpy.denoise.bilateral_filtering import \
+        angle_aware_bilateral_filtering_cpu
+from scilpy.reconst.utils import get_sh_order_and_fullness
+from scilpy.tests.arrays import (
+    fodf_3x3_order8_descoteaux07, fodf_3x3_order8_descoteaux07_filtered)
+
+
+def test_angle_aware_bilateral_filtering_cpu_one_process():
+    """ Test angle_aware_bilateral_filtering_cpu on a simple 3x3 grid
+    using one process.
+    """
+
+    in_sh = fodf_3x3_order8_descoteaux07
+    sh_order = 8
+    sh_basis = 'descoteaux07'
+    in_full_basis = False
+    sphere_str = 'repulsion100'
+    sigma_spatial = 1.0
+    sigma_angular = 1.0
+    sigma_range = 1.0
+    nbr_processes = 1.0
+
+    sh_order, full_basis = get_sh_order_and_fullness(in_sh.shape[-1])
+    out = angle_aware_bilateral_filtering_cpu(in_sh, sh_order,
+                                              sh_basis, in_full_basis,
+                                              sphere_str, sigma_spatial,
+                                              sigma_angular, sigma_range,
+                                              nbr_processes)
+
+    assert np.allclose(out, fodf_3x3_order8_descoteaux07_filtered)
+
+
+def test_angle_aware_bilateral_filtering_cpu_four_processes():
```

Comment:
Do we really prefer having a whole second test with most lines written twice rather than one test_ method with two asserts? Its really your choice, just asking.

Reply:
We can try to parametrize nearly identical test with this approach?
https://github.com/tee-ar-ex/trx-python/blob/master/trx/tests/test_memmap.py
@pytest.mark.parametrize

It basically just expect a list of all parameters combo to give to the test, the test function now has parameters and it runs them all.

You can also generate tests that pass and fail, like the first on in the file I link

Reply:
I agree that I can reduce code duplication in this case, but I would be wary of over-engineering tests. Tests should be simple, simply made and easy to understand, or you can risk introducing bugs in your tests (*gasp*).

Reply:
I agree with Antoine on this. I prefer several simple test cases that check limited subsets of things. To promote code reuse, when can also use pytest fixtures to do setup and teardown of the test cases https://docs.pytest.org/en/6.2.x/fixture.html

Reply:
Addressed in [4ce893b](https://github.com/scilus/scilpy/pull/631/commits/4ce893b4ec7ea8aaa7d8161e8cb875c669f870a3)

Reply:
Yeah, this is perfect like that.